module/Product/src/Controller/ProductController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace Product\Controller;
+
+use Zend\Mvc\Controller\AbstractActionController;
+use Zend\View\Model\ViewModel;
+use Doctrine\ORM\EntityManager;
+use Product\Repository\ProductRepository;
+use Category\Repository\CategoryRepository;
+
+class ProductController extends AbstractActionController
+{
+
+  private $entityManager;
+
+  public function __construct($serviceManager)
+  {
+     $this->entityManager = $serviceManager->get(EntityManager::class);
+  }
+
+  public function indexAction()
+  {
+      $products = new ProductRepository($this->entityManager);
+      $id = (int)$this->params()->fromRoute('id', -1);
+
+
+      if ($id != '-1') {
+        $data = $products->findByCategory($id);
+      } else {
+        $data = $products->findAll();
+      }
+
+      return new ViewModel(['products' => $data]);
+  }
+
+  public function addAction()
+  {
+      $categories = new CategoryRepository($this->entityManager);
+      return new ViewModel(['categories' => $categories->findAll()]);
+  }
+
+  public function addAjaxAction()
+  {
+      if($this->getRequest()->isPost()) {
+         $product = new ProductRepository($this->entityManager);
+         // Fill in the form with POST data
+         $data = $this->params()->fromPost();
+         //var_dump($data); die;
+         $product->create($data);
```

Comment:
I'm seeing a 500 error here when I run this. It appears the `category` isn't being set correctly.